crates/typst/src/eval/ops.rs

```diff
@@ -568,6 +569,28 @@ fn try_cmp_datetimes(a: &Datetime, b: &Datetime) -> StrResult<Ordering> {
         .ok_or_else(|| eco_format!("cannot compare {} and {}", a.kind(), b.kind()))
 }
 
+/// Try to compare arrays of values lexicographically.
+fn try_cmp_arrays(a: &[Value], b: &[Value]) -> StrResult<Ordering> {
+    let cmp_result =
+        a.iter()
+            .zip(b.iter())
+            .try_fold(Ordering::Equal, |accum, (first, second)| {
+                if accum != Ordering::Equal {
+                    Ok(accum)
+                } else {
+                    compare(first, second)
+                }
+            });
+
+    if cmp_result == Ok(Ordering::Equal) {
+        // The two arrays are equal up to the shortest array's extent,
+        // so compare their lengths instead.
+        Ok(a.len().cmp(&b.len()))
+    } else {
+        cmp_result
+    }
+}
```

Comment:
Does this behave differently from [`Iterator::cmp`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.cmp)?

Reply:
I was wondering why you didn't just use `Iterator::cmp`, but I think it understand now. It's because of the StrResult, right?